app/models/invenio_rdm_record_converter.rb

```diff
@@ -16,10 +16,34 @@ def initialize(generic_file=nil)
     return unless generic_file
     @@header_lookup ||= HeaderLookup.new
 
-    #PIDs
-    @pids = invennio_pids(generic_file.doi.shift)
-    @metadata = invenio_metadata(generic_file)
-    @provenance = invenio_provenance(generic_file.proxy_depositor, generic_file.on_behalf_of)
+    @record = record_for_export(generic_file)
+    @file = filename_and_content_path(generic_file)
+  end
+
+  def to_json(options={})
+    options[:except] ||= ["memoized_mesh", "memoized_lcsh"]
+    super
+  end
+
+  private
+
+  def filename_and_content_path(generic_file)
+    {
+      "filename": generic_file.filename,
+      "content_path": generic_file_content_path(generic_file.content.checksum.value)
+    }
+  end
+
+  def generic_file_content_path(checksum)
+    "/#{checksum[0..1]}/#{checksum[2..3]}/#{checksum[4..5]}/#{checksum}" unless !checksum
```

Comment:
This is how contents' paths are generated; by taking the first 6 characters of it's checksum, and dividing it by 3. For example, a file with checksum of "abcd1234", is stored at the following path:
`ab/cd/12/abcd1234`

Reply:
Oh wow, that's nasty. It might be worthwhile to leave a comment here explaining that, but I can't think of any better way to deal with this.